### PR TITLE
Expose status.containerStatuses[*].imageID through Downward API

### PR DESCRIPTION
**Pull request by mitar**
See: https://github.com/kubernetes/enhancements/issues/2012

### CODE REVIEW

File: keps/sig-node/2012-expose-image-id-through-downward-api/kep.yaml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  beta: "v1.21"
+  stable: "v1.22"
+
+# The following PRR answers are required at alpha release
+# List the feature gate name and the components for which it must be enabled
+feature-gates:
+  - name: ImageIDDownwardAPI
+    components:
+      - kube-apiserver
+      - kube-controller-manager
```
Comment (mitar) [inline keps/sig-node/2012-expose-image-id-through-downward-api/kep.yaml:35]: I am not sure which components this touch, help here would be appreciated.

File: keps/sig-node/2012-expose-image-id-through-downward-api/kep.yaml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+disable-supported: true
```
Comment (mitar) [inline keps/sig-node/2012-expose-image-id-through-downward-api/kep.yaml:36]: What is this flag? `disable-supported`

**Comment by justincormack** [comment]
If you are going to expose imageID why not expose other information, in particular for the purposes you describe, image lables could be useful too, as these may well contain (as per OCI standard labels) git commit info etc that are also useful here.

**Comment by mitar** [comment]
> If you are going to expose imageID why not expose other information, in particular for the purposes you describe, image lables could be useful too, as these may well contain (as per OCI standard labels) git commit info etc that are also useful here.

I am open to that, but it is not my use case. Having imageID I can retrieve image labels at a later time. Also imageID is already readily available in Kubernetes while image labels I do not think they are? But yes, I am open to that as well.

**Comment by mitar** [comment]
@mrunalp Any update on the review?

**Comment by mitar** [comment]
OK, so the error is that it is still unclear who would approve it?

**Comment by ehashman** [comment]
@mitar you need to add a PRR approver as well since this is net new. See https://github.com/kubernetes/community/blob/master/sig-architecture/production-readiness.md#submitting-a-kep-for-production-readiness-approval

File: keps/sig-node/2012-expose-image-id-through-downward-api/README.md [hunk 3]
```diff
@@ -0,0 +1,497 @@
...
+  cluster required to make on upgrade, in order to maintain previous behavior?
+- What changes (in invocations, configurations, API use, etc.) is an existing
+  cluster required to make on upgrade, in order to make use of the enhancement?
+-->
+
+### Version Skew Strategy
+
+Implementation of this is contained to only the executor component for containers,
+which has to pass on the digest of the image it is using to run the container.
+As such version skew is not anticipated to cause issues.
```
Comment (BenTheElder) [inline keps/sig-node/2012-expose-image-id-through-downward-api/README.md:280]: Is it actually though?
IIRC pod validation happens in the api-server, but populating this downward API value would be in kubelet?

File: keps/sig-node/2012-expose-image-id-through-downward-api/kep.yaml (hunk 1 above)
Comment (BenTheElder) [reply keps/sig-node/2012-expose-image-id-through-downward-api/kep.yaml:35]: kubelet?

File: keps/sig-node/2012-expose-image-id-through-downward-api/kep.yaml (hunk 2 above)
Comment (BenTheElder) [reply keps/sig-node/2012-expose-image-id-through-downward-api/kep.yaml:36]: wether the feature can be disabled?

**Comment by mitar** [comment]
Sorry, I do not have time to pursue this further.

**Comment by PringBlocks** [comment]
how far away was this PR from getting merged? I'm interested in helping out if that's an option

**Comment by mitar** [comment]
@PringBlocks I think it was mostly missing reviews. Feel free to reuse it and reopen it.

**Comment by svrnm** [comment]
@PringBlocks , @mitar any interest on reviving this? I am more interested in the containerID, but I think they go well together

**Comment by mitar** [comment]
I still think it would be useful, but I sadly do not have time to work on it anymore.

**Comment by mitar** [comment]
Sure. I mean, you could fork the branch and create a new PR. I think that would be the cleanest.

**Comment by svrnm** [comment]
> Sure. I mean, you could fork the branch and create a new PR. I think that would be the cleanest.

thanks, I will see how I can go about it

**Pull request closed without merging**